sdk/metric/meter.go

```diff
@@ -440,52 +450,58 @@ func (m *meter) RegisterCallback(f metric.Callback, insts ...metric.Observable)
 		// Don't allocate a observer if not needed.
 		return noopRegister{}, nil
 	}
-
-	reg := newObserver()
+	unregs := make([]func(), len(m.pipes))
 	var err error
-	for _, inst := range insts {
-		switch o := inst.(type) {
-		case int64Observable:
-			if e := o.registerable(m); e != nil {
-				if !errors.Is(e, errEmptyAgg) {
-					err = errors.Join(err, e)
+	for ix, pipe := range m.pipes {
+		reg := newObserver(pipe)
+		for _, inst := range insts {
+			switch o := inst.(type) {
+			case int64Observable:
+				if e := o.registerable(m); e != nil {
+					if !errors.Is(e, errEmptyAgg) {
```

Comment:
I think this will return the same error multiple times (once for each pipeline).  I think we should iterate over instruments once to see if it is registerable, and construct a list of valid instruments.  Then iterate over valid instruments once for each pipe and register the valid ones.

Reply:
makes sense. I will make the change

Reply:
@dashpole pushed the change